### PR TITLE
Improve docs for `AdminServlet` et al

### LINKED ISSUE #1896: Create documentation for #1890
Issue #1890 added new configuration to the `AdminServlet` to allow disabling of the individual features. These changes need to be documented.

### PR DESCRIPTION
**Pull request by mschechter-bellese**
Added documentation for the `AdminServlet` and the other metrics-provided servlets, including the new configuration added as part of #1890.

Resolves #1896

### CODE REVIEW

File: docs/source/manual/servlets.rst [hunk 1]
```diff
@@ -14,12 +14,52 @@ HealthCheckServlet
 ``HealthCheckServlet`` responds to ``GET`` requests by running all the [health checks](#health-checks)
 and returning ``501 Not Implemented`` if no health checks are registered, ``200 OK`` if all pass, or
 ``500 Internal Service Error`` if one or more fail. The results are returned as a human-readable
-``text/plain`` entity.
+``application/json`` entity.
```
Comment (joschi) [inline docs/source/manual/servlets.rst]: Maybe we could just write "as a human-readable JSON entity"?
```suggestion
JSON entity.
```

File: docs/source/manual/servlets.rst [hunk 2]
```diff
@@ -14,12 +14,52 @@ HealthCheckServlet
...
+application is healthy. If not provided, it defaults to ``"true"``
+
+Query Parameters
+~~~~~~~~~~~~~~~~
+
+``HealthCheckServlet`` supports the following query parameters:
+
+* ``httpStatusIndicator`` (`Boolean``): Determines whether the HTTP status code is used to
+  determine whether the application is healthy; if not provided, it defaults to the value from the
+  initialization parameter
```
Comment (joschi) [inline docs/source/manual/servlets.rst]: Could we describe the possible HTTP response codes (200, 500, 501)?

**Review by joschi — COMMENTED** [review]
@mschechter-bellese Thanks for the awesome contribution! ❤️

File: docs/source/manual/servlets.rst (hunk 1 above)
Comment (mschechter-bellese) [reply docs/source/manual/servlets.rst]: Updated - thanks!

File: docs/source/manual/servlets.rst (hunk 2 above)
Comment (mschechter-bellese) [reply docs/source/manual/servlets.rst]: They're in the lead paragraph; I split them out for clarity

**Review by joschi — APPROVED** [review]
@mschechter-bellese Thanks a lot! ❤️

**Comment by mschechter-bellese** [comment]
> @mschechter-bellese Thanks a lot! ❤️

You're welcome - glad to help out! I probably should have done this on my previous issue :) 

Next up is getting these changes integrated into Dropwizard via `AbstractServerFactory`...

**Pull request merged**